### ISSUE TITLE
Zoom and pan features should work in Dependency Map and Trace View

### ISSUE THREAD

**Issue by ahtrotta** [getappmap/appmap-js#1206]
If using `@appland/components` version `0.44.0` or greater with the VS Code extension, then the pan and zoom functions do not currently work in the Dependency Map or the Trace View. If you open either of those tabs then you get this console error:

![Image](https://github.com/getappmap/appmap-js/assets/45714532/f650787b-eab9-4a71-b2e3-a487c121c721)

It looks like it might have something to do with the different versions of the d3 library in use by `diagrams` and `components` since the stack trace references [`centerX`](https://github.com/getappmap/appmap-js/blob/main/packages/diagrams/src/helpers/container/index.js#L162), which pulls in `zoomIdentity` from `d3-zoom` [here](https://github.com/getappmap/appmap-js/blob/main/packages/diagrams/src/helpers/container/index.js#L2).

**TL;DR: It looks like it's because we have [more than one instance of the `d3-selection` module in our dependency tree](https://github.com/vasturiano/react-force-graph/issues/310#issuecomment-892216490).**

**Comment by lachrist** [comment]
`d3-selection` is required in `d3-flame-graph` as `3.0.0`:

```
├─ d3-flame-graph@npm:4.1.3
│  ├─ Version: 4.1.3
│  │
│  └─ Dependencies
│     ├─ d3-array@npm:^3.1.1 → npm:3.2.3
│     ├─ d3-dispatch@npm:^3.0.1 → npm:3.0.1
│     ├─ d3-ease@npm:^3.0.1 → npm:3.0.1
│     ├─ d3-format@npm:^3.0.1 → npm:3.1.0
│     ├─ d3-hierarchy@npm:^3.0.1 → npm:3.1.2
│     ├─ d3-scale@npm:^4.0.2 → npm:4.0.2
│     ├─ d3-selection@npm:^3.0.0 → npm:3.0.0
│     └─ d3-transition@npm:^3.0.1 → npm:3.0.1 [10ac6]
```

And as `1.4.2` in `@appland/diagrams`:

```
├─ @appland/diagrams@portal:/Users/lachrist/Desktop/workspace/appmap-js/packages/diagrams::locator=appmap%40workspace%3A.
│  ├─ Version: 1.6.3
│  │
│  └─ Dependencies
│     ├─ @appland/models@portal:/Users/lachrist/Desktop/workspace/appmap-js/packages/models::locator=appmap%40workspace%3A. → portal:/Users/lachrist/Desktop/workspace/appmap-js/packages/models::locator=appmap%40workspace%3A.
│     ├─ d3-interpolate@npm:^1.4.0 → npm:1.4.0
│     ├─ d3-selection@npm:^1.4.2 → npm:1.4.2
│     ├─ d3-shape@npm:^1.3.7 → npm:1.3.7
│     ├─ d3-zoom@npm:^1.8.3 → npm:1.8.3
│     ├─ dagre@npm:^0.8.5 → npm:0.8.5
│     └─ deepmerge@npm:^4.2.2 → npm:4.2.2
```

**Comment by lachrist** [comment]
Here is our option to solve this:
- Use a previous version `d3-flamegraph` which rely on `d3-selection@1`.
- Use `d3-selection@3` in `@appland/diagrams`.

This first option seems risky because `d3-flamegraph` is not that well maintained and I would not want to be stuck with a legacy version of this. The second option however might require non-trivial changes in `@appland/diagram`. Since `d3-selection@2` is dropping support for the global export `event`.

https://github.com/d3/d3-selection/releases/tag/v2.0.0
